action_server/src/robocorp/action_server/_server_expose.py

```diff
@@ -15,6 +16,25 @@
 log = logging.getLogger(__name__)
 
 
+# websockets logger will log the retries with the stack trace. this makes it less verbose.
+class NoExceptionFormatter(logging.Formatter):
+    def format(self, record):
+        record.exc_info = None
+        return super().format(record)
+
+
+console_handler = logging.StreamHandler()
+console_handler.setLevel(logging.DEBUG)
+
+formatter = NoExceptionFormatter("%(message)s")
+console_handler.setFormatter(formatter)
+
+
+websockets_logger = logging.getLogger("websockets")
+websockets_logger.addHandler(console_handler)
```

Comment:
I think this shouldn't be done here... the logging is configured at `robocorp.action_server.cli._setup_stdout_logging` and `robocorp.action_server.cli._setup_logging`, so, this probably belongs there...

Reply:
`websockets` has its own re-connect logic, but it prints the failure logs with the stack trace making the output too verbose. This is just a workaround to stop it from printing the stack trace. See [websockets code](https://github.com/python-websockets/websockets/blob/35bc7dd8288445289134c335aae8af859862ccd1/src/websockets/legacy/client.py#L606) 

So I don't think this workaround should be moved to `robocorp.action_server.cli`.

Reply:
The issue here is that the code is adding directly a logger and stopping propagation and the logger is redirected to stdout.

The action server actually adds 2 handlers, one for the stdout and another to a rotating file handler. The code here just adds one related to the stdout and stops the propagation for the root in the stdout, but then the rotating file handler doesn't get any notification.

This means that logs will always be shown in the console the way it's done here and not in the file and the log level here doesn't respect the log level that was configured in `_setup_stdout_logging`/`_setup_logging`, which is not what's expected (it's expected that the log level will respect the user setting and also it should redirect to both, stdout and the file).

Reply:
Okay, any pointers how to
1. keep `websockets` re-connect logging
2. remove the stack trace
without my changes?

Reply:
Maybe you could change the current formatters to format differently (i.e.: remove the exception info) based on the `module` and `funcName` available?  (in `_setup_stdout_logging`/`_setup_logging`)

i.e.: Create a new formatter which wraps an existing formatter and just implements `format` and removes the exception info if the record module/funcName match something you expect. That way you don't need to add a new handler, just change the existing formatters to be that formatter-wrapper.

Does it make sense?